baselines/fedexp/fedexp/models.py

```diff
@@ -0,0 +1,184 @@
+
+    Parameters
+    ----------
+    net : nn.Module
+        The neural network to test.
+    test_loader : DataLoader
+        The DataLoader containing the data to test the network on.
+    device : torch.device
+        The device on which the model should be tested, either 'cpu' or 'cuda'.
+
+    Returns
+    -------
+    Tuple[float, float]
+        The loss and the accuracy of the input model on the given data.
+    """
+    if len(test_loader.dataset) == 0:
+        raise ValueError("Testloader can't be 0, exiting...")
+
+    criterion = torch.nn.CrossEntropyLoss()
+    correct, total, loss = 0, 0, 0.0
+    net.eval()
+    with torch.no_grad():
+        for images, labels in tqdm(test_loader, "Testing ..."):
+            images, labels = images.to(device), labels.to(device)
+            outputs = net(images)
+            loss += criterion(outputs, labels).item()
+            _, predicted = torch.max(outputs.data, 1)
+            total += labels.size(0)
+            correct += (predicted == labels).sum().item()
+    loss /= len(test_loader.dataset)
+    accuracy = correct / total
+    return loss, accuracy
+
+
+def train(  # pylint: disable=too-many-arguments
+        net: nn.Module,
+        trainloader: DataLoader,
+        device: torch.device,
+        epochs: int,
+        hyperparams: Dict[str, Scalar],
+) -> None:
+    """Train the network on the training set.
+
+    Parameters
+    ----------
+    net : nn.Module
+        The neural network to train.
+    trainloader : DataLoader
+        The DataLoader containing the data to train the network on.
+    device : torch.device
+        The device on which the model should be trained, either 'cpu' or 'cuda'.
+    epochs : int
+        The number of epochs the model should be trained for.
+    hyperparams : Dict[str, Scalar]
+        The hyperparameters to use for training.
+    """
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(
+        net.parameters(),
+        lr=hyperparams["eta_l"],
+        momentum=0,
+        weight_decay=hyperparams["weight_decay"],
+    )
+    net.train()
+    counter = 0
+    for images, labels in trainloader:
+        images, labels = images.to(device), labels.to(device)
+        if hyperparams["use_data_augmentation"]:
+            transform_train = transforms.Compose(
+                [
+                    transforms.RandomCrop(32, padding=4),
+                    transforms.RandomHorizontalFlip(),
+                ]
+            )
+            images = transform_train(images)
+        net.zero_grad()
+        log_probs = net(images)
+        loss = criterion(log_probs, labels)
+        loss.backward()
+        if hyperparams["use_gradient_clipping"]:
+            torch.nn.utils.clip_grad_norm_(
+                parameters=net.parameters(), max_norm=hyperparams["max_norm"]
+            )
+        optimizer.step()
+        counter += 1
+        if counter >= epochs:
+            break
```

Comment:
This is not used anymore since you moved it to the client's `fit()`. Can we remove this then?